src/app/components/SubjectHeading/SubjectHeadingSearch.jsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { Link } from 'react-router';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+import appConfig from '../../../../appConfig';
+import AutosuggestItem from './AutosuggestItem'
+
+class SubjectHeadingSearch extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activeSuggestion: 0,
+      suggestions: [],
+      userInput: ''
+    }
+
+    this.onChange = this.onChange.bind(this)
+    this.onSubmit = this.onSubmit.bind(this)
+    this.clearUserInput = this.clearUserInput.bind(this)
+    this.generatePath = this.generatePath.bind(this)
+  }
+
+  onSubmit(submitEvent) {
+    submitEvent.preventDefault();
+    const url = this.generatePath(this.state.suggestions[this.state.activeSuggestion])
+    console.log(this.props, this.context);
+    this.clearUserInput()
+    this.context.router.push(url)
+  }
+
+  onChange(inputEvent) {
+    const { suggestion } = this.state
+    const userInput = inputEvent.currentTarget.value
+
+    this.setState({
+      userInput
+    })
+
+    if (userInput.length > 2) {
+      axios({
+        method: 'GET',
+        url: `${appConfig.shepApi}/autosuggest?query=${userInput}`,
+        crossDomain: true,
+        headers: {
+          'Access-Control-Allow-Origin': '*',
```

Comment:
Is this necessary? I thought this header was used only by a server to strongly suggest what origins may request resources? I'm not sure what meaning it has as a request header

Reply:
Does not seem to be.